sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ManagementChannel.java

```diff
@@ -74,6 +88,16 @@
             "'cbsBasedTokenManager' cannot be null.");
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<Instant> renewMessageLock(ServiceBusReceivedMessage messageForLockRenew) {
+        return renewMessageLock(new ServiceBusReceivedMessage[]{messageForLockRenew})
+            .elementAt(0)
```

Comment:
Why element at? And why forcing it on another scheduler?

Reply:
This should use `Flux.next` instead. Right now we have the possibility that the `Flux` could be empty and `elementAt` of 0 will throw an `IndexOutOfBounds` exception due to this, with `next` it will return an empty `Mono`.